includes/class-signature.php

```diff
@@ -186,7 +186,11 @@ protected static function check_legacy_key_pair_for( $user_id ) {
 	 */
 	public static function generate_signature( $user_id, $http_method, $url, $date, $digest = null ) {
 		$user = Users::get_by_id( $user_id );
-		$key  = self::get_private_key_for( $user->get__id() );
+		if ( ! is_wp_error( $user ) ) {
+			$key = self::get_private_key_for( $user->get__id() );
+		} else {
+			$key = get_option( 'activitypub_temp_sig_' . $user_id, true );
```

Comment:
this is not the user-id! see: https://github.com/mediaformat/wordpress-activitypub/blob/cf5e95eb07a4044e28481e7f15982c07e84b027b/includes/class-signature.php#L131C28-L131C57

Reply:
l'd like to reuse this however there is a [race condition](https://github.com/Automattic/wordpress-activitypub/pull/552#issuecomment-1806592481) preventing the use of $user object at this stage.
https://github.com/Automattic/wordpress-activitypub/blob/f6b4783dedbc5ad07c42a9c49a275ecaed05a227/includes/class-signature.php#L131-L137

Hence the use of a temporary option to store the signature before user deletion.